include/screencast_common.h

```diff
@@ -91,6 +94,10 @@ struct xdpw_screencast_instance {
 	bool with_cursor;
 	int err;
 	bool quit;
+
+	// fps limit
+	double max_fps; // 0 means no limit
```

Comment:
We don't need to store this in two places. If it is currently implemented for all screencasts, it isn't needed here.

Reply:
The reason I added `max_fps` here is so that I could use it in `wlr_frame_buffer_done`:

```C
static void wlr_frame_buffer_done(void *data,
		struct zwlr_screencopy_frame_v1 *frame) {
	struct xdpw_screencast_instance *cast = data;

	logprint(TRACE, "wlroots: buffer_done event handler");

	fps_limit_apply(&cast->fps_limit, cast->max_fps);
...
```

Since this is a callback, I presume the only way to pass information to it is to use include it in the `void *data` argument. How am I supposed to pass this information here otherwise? The only other way I can think of is to introduce a global variable, but I am not sure if that's a good idea.

Reply:
The screencast context is [available from any screencast instance](https://github.com/emersion/xdg-desktop-portal-wlr/blob/e103e120e20e83acec5b66ce4cd33a0eb86df0b0/include/screencast_common.h#L73) via
```c
cast->ctx;
```

Doing it this way keeps us from maintaining this information in more than one place.

Reply:
You are right, I completely missed the context being already there.